doc/developer_guide.md

```diff
@@ -0,0 +1,213 @@
+    qflags\test_Qt3DCore_DeliveryFlags_DeliveryFlag.py::test_on_one_flag_class PASSED              [  5%]
+    qflags\test_Qt3DCore_DeliveryFlags_DeliveryFlag.py::test_on_multi_flag_class PASSED            [  6%]
+    [...]
+    qflags\test_QtWidgets_ViewItemFeatures_ViewItemFeature.py::test_on_one_flag_class PASSED       [ 99%]
+    qflags\test_QtWidgets_ViewItemFeatures_ViewItemFeature.py::test_on_multi_flag_class PASSED     [ 99%]
+    qflags\test_QtWidgets_WizardOptions_WizardOption.py::test_on_one_flag_class PASSED             [ 99%]
+    qflags\test_QtWidgets_WizardOptions_WizardOption.py::test_on_multi_flag_class PASSED           [100%]
+
+    ======================================= 382 passed in 11.72s ========================================
+
+
+
+The different phases of the testing process are driven by the script `test_stub.py`. The steps are:
+
+1. Run mypy on every file of the directory not starting with `test_` . This is the part which generates
+   the output:
+
+```
+   test_stubs.py::test_stubs[pyqtsignal.py] PASSED                                                [  0%]
+   test_stubs.py::test_stubs[pyqtslot.py] PASSED                                                  [  0%]
+   [...]
+   test_stubs.py::test_files[qtreewidgetitem.py] PASSED                                           [  4%]
+   test_stubs.py::test_files[simple.py] PASSED                                                    [  4%]
+```
+   
+2. Run the example python files themselves, to make sure they are correct. This is
+   the longest part of the output:
+
+```
+   qflags\test_Qt3DCore_ChangeFlags_ChangeFlag.py::test_on_one_flag_class PASSED                  [  5%]
+   qflags\test_Qt3DCore_ChangeFlags_ChangeFlag.py::test_on_multi_flag_class PASSED                [  5%]
+   [...]
+   qflags\test_QtWidgets_ViewItemFeatures_ViewItemFeature.py::test_on_one_flag_class PASSED       [ 99%]
+   qflags\test_QtWidgets_ViewItemFeatures_ViewItemFeature.py::test_on_multi_flag_class PASSED     [ 99%]
+
+   ======================================= 382 passed in 11.72s ========================================
+```
+
+
+### Adding new tests
+
+Each time you find an incorrect stub, it is a good idea to add a dedicated test to show
+what is expected from mypy. The steps to take are:
+
+1. Add a new file under the test directory with a meaningful name (usually, the name of the 
+   relevant class). Example: `qlineedit.py`
+   
+2. In the file, write the correct python code which should typecheck correctly. This is usually
+    just a few lines of code as you can see from the other examples. A QApplication has already 
+    been created as part of the test execution, so you are free to create an QObject or widgets.
+   
+3. Run your example file with Python. There should be no error and no output. 
+   
+   Example:
+
+```
+   (env_for_tests) .../PyQt5-stubs/tests/$ python qlineedit.py
+   (env_for_tests) .../PyQt5-stubs/tests/$
```

Comment:
No need to indent when using ```.

```suggestion
(env_for_tests) .../PyQt5-stubs/tests/$ python qlineedit.py
(env_for_tests) .../PyQt5-stubs/tests/$
```